src/test/scala/uk/gov/nationalarchives/tdr/validation/MetadataValidationJsonSchemaSpec.scala

```diff
@@ -0,0 +1,133 @@
+package uk.gov.nationalarchives.tdr.validation
+
+import org.apache.pekko.actor.ActorSystem
+import org.apache.pekko.testkit.{ImplicitSender, TestKit}
+import org.scalatest.matchers.should.Matchers._
+import org.scalatest.wordspec.AnyWordSpecLike
+import uk.gov.nationalarchives.tdr.validation.schema.JsonSchemaDefinition.BASE_SCHEMA
+import uk.gov.nationalarchives.tdr.validation.schema.MetadataValidationJsonSchema
+import uk.gov.nationalarchives.tdr.validation.schema.MetadataValidationJsonSchema.ObjectMetadata
+
+class MetadataValidationJsonSchemaSpec extends TestKit(ActorSystem("MySpec")) with ImplicitSender with AnyWordSpecLike {
+
+  "MetadataValidationJsonSchema" should {
+    "validate incorrect value in enumerated array" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Language", "Unknown")
+      val validationErrors = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 1
+      singleErrorCheck(validationErrors, "language", "enum")
+    }
+    "validate correct value enumerated array" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Language", "Welsh")
+      val validationErrors = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 0
+    }
+    "validate array can be null" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Language", "")
+      val validationErrors = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 0
+    }
+    "validate incorrect date" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Date last modified", "12-12-2012")
+      val validationErrors: Map[String, List[Error]] = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 1
+    }
+    "validate correct date" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Date last modified", "2023-12-05")
+      val validationErrors = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 0
+      singleErrorCheck(validationErrors, "date_last_modified", "format.date")
+    }
+    "validate date last modified must have a value" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Date last modified", "")
+      val validationErrors = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 1
+      singleErrorCheck(validationErrors, "date_last_modified", "type")
+    }
+    "validate end date can be empty" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Date of the record", "")
+      val validationErrors = MetadataValidationJsonSchema.validate(BASE_SCHEMA, data)
+      validationErrors("file1").size shouldBe 0
+    }
+    "validate end date must be before today" in {
+      val data: Set[ObjectMetadata] = dataBuilder("Date of the record", "2044-12-12")
```

Comment:
Might want to take the date before today to something we're ever likely to see, eg "3000-12-12"